packages/common/src/utils/formatting.test.ts

```diff
@@ -1,412 +1,883 @@
-            expect(formatValue(10, roundZeroOptions)).toEqual('10');
-            expect(formatValue(100, roundZeroOptions)).toEqual('100');
-            expect(formatValue(1000, roundZeroOptions)).toEqual('1,000');
-            expect(formatValue(10000, roundZeroOptions)).toEqual('10,000');
-            expect(formatValue(100000, roundZeroOptions)).toEqual('100,000');
-            expect(formatValue(1000000, roundZeroOptions)).toEqual('1,000,000');
-            expect(formatValue(5, roundZeroOptions)).toEqual('5');
-            expect(formatValue(5.001, roundZeroOptions)).toEqual('5');
-            expect(formatValue(5.9999999, roundZeroOptions)).toEqual('6');
-
-            const roundTwoOptions = {
-                round: 2,
-            };
-            expect(formatValue(5, roundTwoOptions)).toEqual('5.00');
-            expect(formatValue(5.001, roundTwoOptions)).toEqual('5.00');
-            expect(formatValue(5.555, roundTwoOptions)).toEqual('5.56');
-            expect(formatValue(5.5555, roundTwoOptions)).toEqual('5.56');
-            expect(formatValue(5.9999999, roundTwoOptions)).toEqual('6.00');
-            expect(formatValue('foo', roundTwoOptions)).toEqual('foo');
-            expect(formatValue(false, roundTwoOptions)).toEqual('false');
-
-            const roundTenOptions = {
-                round: 10,
-            };
-            expect(formatValue(5, roundTenOptions)).toEqual('5.0000000000');
-            expect(formatValue(5.001, roundTenOptions)).toEqual('5.0010000000');
-            expect(formatValue(5.9999999, roundTenOptions)).toEqual(
-                '5.9999999000',
-            );
 
-            // negative rounding not supported
-            expect(
-                formatValue(5, {
-                    round: -1,
-                }),
-            ).toEqual('5');
+        describe(`when it is ${Format.PERCENT.toUpperCase()}`, () => {
+            test('getCustomFormatFromLegacy should return the correct CustomFormat options', () => {
+                expect(
+                    getCustomFormatFromLegacy({ format: Format.PERCENT }),
+                ).toEqual({
+                    type: CustomFormatType.PERCENT,
+                    compact: undefined,
+                    round: undefined,
+                });
+            });
+        });
+
+        describe(`when it is ${Format.ID.toUpperCase()}`, () => {
+            test('getCustomFormatFromLegacy should return the correct CustomFormat options', () => {
+                expect(
+                    getCustomFormatFromLegacy({ format: Format.ID }),
+                ).toEqual({
+                    type: CustomFormatType.ID,
+                });
+            });
+        });
+    });
+
+    describe('applying CustomFormat to value', () => {
+        describe('when using legacy format', () => {
+            describe('if Format is distance unit', () => {
+                test('it should return the right format', () => {
```

Comment:
nit-pick: can just have one `test` and remove the extra `describe` in `71` 
```
test('returns formatted value if distance-formatting') 
```
or something similar 😅